nemo/utils/exp_manager.py

```diff
@@ -565,69 +558,70 @@ def check_resume(
+                elif checkpoint == resume_from_checkpoint:
+                    warn += f"Training from {resume_from_checkpoint}."
+                logging.warning(warn)
+            else:
+                raise NotFoundError(
+                    f"There were no checkpoints found in checkpoint_dir or no checkpoint folder at checkpoint_dir :{checkpoint_dir}. Cannot resume."
+                )
+        elif len(end_checkpoints) > 0:
+            if resume_past_end:
+                if len(end_checkpoints) > 1:
+                    if 'mp_rank' in str(end_checkpoints[0]):
+                        checkpoint = end_checkpoints[0]
+                    else:
+                        raise ValueError(f"Multiple checkpoints {end_checkpoints} that matches *end.ckpt.")
+            else:
+                raise ValueError(
+                    f"Found {end_checkpoints[0]} indicating that the last training run has already completed."
+                )
+        elif len(last_checkpoints) > 1:
+            if 'mp_rank' in str(last_checkpoints[0]) or 'tp_rank' in str(last_checkpoints[0]):
+                checkpoint = last_checkpoints[0]
+                checkpoint = uninject_model_parallel_rank(checkpoint)
+            else:
+                raise ValueError(f"Multiple checkpoints {last_checkpoints} that matches *last.ckpt.")
         else:
-            raise NotFoundError(f"There were no checkpoints found in {checkpoint_dir}. Cannot resume.")
-    elif len(last_checkpoints) > 1:
-        if 'mp_rank' in str(last_checkpoints[0]) or 'tp_rank' in str(last_checkpoints[0]):
             checkpoint = last_checkpoints[0]
-            checkpoint = uninject_model_parallel_rank(checkpoint)
-        else:
-            raise ValueError(f"Multiple checkpoints {last_checkpoints} that matches *last.ckpt.")
-    else:
-        logging.info(f"Resuming from {last_checkpoints[0]}")
-        checkpoint = last_checkpoints[0]
 
     # PTL 2.0 supports ckpt_path instead of resume_from_checkpoint as the trainer flag
-    trainer.ckpt_path = str(checkpoint)
-
-    if is_global_rank_zero():
-        # Check to see if any files exist that need to be moved
-        files_to_move = []
-        for child in Path(log_dir).iterdir():
-            if child.is_file():
-                files_to_move.append(child)
-
-        if len(files_to_move) > 0:
-            # Move old files to a new folder
-            other_run_dirs = Path(log_dir).glob("run_*")
-            run_count = 0
-            for fold in other_run_dirs:
-                if fold.is_dir():
-                    run_count += 1
-            new_run_dir = Path(Path(log_dir) / f"run_{run_count}")
-            new_run_dir.mkdir()
-            for _file in files_to_move:
-                move(str(_file), str(new_run_dir))
+    if checkpoint is not None:
+        trainer.ckpt_path = str(checkpoint)
+        logging.info(f'Resuming training from checkpoint: {trainer.ckpt_path}')
+
+        if is_global_rank_zero():
```

Comment:
This should be moved left indent one time - earlier code has it outside.

Reply:
Earlier code also had an early return when checkpoint dir doesn't exist but I had to remove that for `resume_from_checkpoint`. So I'm not sure if this section should run when `trainer.ckpt_path` is `None`.